src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/EmbeddingProviderResponseValidation.java

```diff
@@ -36,20 +46,33 @@ public void filter(ClientRequestContext requestContext, ClientResponseContext re
     if (responseContext.getStatus() == 0) {
       return;
     }
+    // Throw error if there is no response body
+    if (!responseContext.hasEntity()) {
+      throw EMBEDDING_PROVIDER_UNEXPECTED_RESPONSE.toApiException(
+          "No response body from the embedding provider");
+    }
+
     // Check the Content-Type of the response
     MediaType contentType = responseContext.getMediaType();
     if (contentType == null
         || !(MediaType.APPLICATION_JSON_TYPE.isCompatible(contentType)
             || MEDIATYPE_TEXT_JSON.isCompatible(contentType))) {
```

Comment:
Just to make sure I understand: is response only read in case it is not declared as JSON?
If so, makes sense. But let's change above "Check the Content-Type of the response" to something that indicates what and why is done here.
(basically: response should always be JSON; if not, error out, include raw response message for trouble-shooting purposes)

Reply:
Yes, correct. Will update the comments